flw/src/main/java/org/motechproject/nms/flw/service/impl/FrontLineWorkerImportServiceImpl.java

```diff
@@ -87,7 +88,11 @@ public void importFrontLineWorker(Map<String, Object> record, State state) throw
         if (flw == null) {
             frontLineWorkerService.add(createFlw(record, location));
         } else {
-            frontLineWorkerService.update(updateFlw(flw, record, location));
+            LocalDate mctsUpdatedDateNic = (LocalDate) record.get(FlwConstants.UPDATED_ON);
+            //It updated_date_nic from mcts is not null,then it's not a new record. Compare it with the record from database and update
+            if (mctsUpdatedDateNic != null && (flw.getUpdatedDateNic() == null || mctsUpdatedDateNic.isAfter(flw.getUpdatedDateNic()) || mctsUpdatedDateNic.isEqual(flw.getUpdatedDateNic()))) {
+                frontLineWorkerService.update(updateFlw(flw, record, location));
```

Comment:
I will add a logger here. Else case is for records with older data. If updated ones are imported we ignore the older records from the same file. 
This case can occur if there are any failed imports in the past 7 days